src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,208 @@
+ */
+public class TaskManager {
+    // Success Messages
+    private static final String TASKS_COUNT_MESSAGE = "Now you have %d %s in the list.";
+    private static final String UNDONE_TASKS_COUNT_MESSAGE = "You have %d incomplete %s remaining.";
+    private static final String TASK_ADDED_MESSAGE = "Got it. I've added this task:\n  %s\n\n" + TASKS_COUNT_MESSAGE;
+    private static final String MARKED_TASK_AS_DONE_MESSAGE = "Nice! I've marked this task as done:\n  %s\n\n"
+            + UNDONE_TASKS_COUNT_MESSAGE;
+    private static final String DELETED_TASK_MESSAGE = "Noted. I've removed this task:\n  %s\n\n" + TASKS_COUNT_MESSAGE;
+
+    // Error Messages
+    private static final String TASK_NOT_FOUND_MESSAGE =
+            "You don't have a task with that number.";
+
+    private final List<Task> taskList;
+
+    public TaskManager(List<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    public TaskManager() {
+        taskList = new ArrayList<>();
+    }
+
+    /**
+     * Returns the number of tasks.
+     */
+    public int getTaskCount() {
+        return taskList.size();
+    }
+
+    /**
+     * Returns the number of incomplete tasks.
+     */
+    private int getUndoneTaskCount() {
+        int count = 0;
+        for (Task t : taskList) {
+            if (!t.checkTaskDone()) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    /**
+     * Adds a <code>Task</code> into the list of tasks.
+     * @param task the task to be added
+     * @return a message containing the task just added and the updated number of tasks
+     */
+    public String addTask(Task task) {
+        taskList.add(task);
+        int taskCount = getTaskCount();
+        String pluralised = taskCount > 1 ? "tasks" : "task";
+        return String.format(TASK_ADDED_MESSAGE, task, taskCount, pluralised);
+    }
+
+    /**
+     * Marks a <code>Task</code> as completed.
+     * @param taskNumber the number of the task to be marked as completed
+     * @return a message containing the task just marked as completed and the updated number of incomplete tasks
+     * @throws DukeException if a task with that number cannot be found
+     */
+    public String markTaskAsDone(int taskNumber) throws DukeException {
+        try {
+            // User input is 1-indexed
+            int taskIndex = taskNumber - 1;
+            Task task = taskList.get(taskIndex);
+            task.markAsDone();
+            int undoneTaskCount = getUndoneTaskCount();
+            String pluralised = undoneTaskCount > 1 || undoneTaskCount == 0 ? "tasks" : "task";
+            return String.format(MARKED_TASK_AS_DONE_MESSAGE, task, undoneTaskCount, pluralised);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException(TASK_NOT_FOUND_MESSAGE);
+        }
+    }
+
+    /**
+     * Deletes a <code>Task</code> from the list.
+     * @param taskNumber the number of the task to be deleted
```

Comment:
I think a punctuation is required behind the parameter description?

Reply:
Yup you're right. Have fixed this!